source/patterns/@aws-solutions-constructs/aws-apigateway-sqs/test/apigateway-sqs.test.ts

```diff
@@ -261,4 +261,238 @@ test('Queue is encrypted with customer managed KMS Key when enable encryption fl
       ]
     }
   });
+});
+
+test('Construct accepts additional read request templates', () => {
+  const stack = new Stack();
+  new ApiGatewayToSqs(stack, 'api-gateway-sqs', {
+    enableEncryptionWithCustomerManagedKey: true,
+    additionalReadRequestTemplates: {
+      'text/plain': 'Hello'
+    }
+  });
+
+  expect(stack).toHaveResourceLike('AWS::ApiGateway::Method', {
+    HttpMethod: 'GET',
+    Integration: {
+      RequestTemplates: {
+        'application/json': 'Action=ReceiveMessage',
+        'text/plain': 'Hello'
+      }
+    }
+  });
+});
+
+test('Construct accepts additional create request templates', () => {
+  const stack = new Stack();
+  new ApiGatewayToSqs(stack, 'api-gateway-sqs', {
+    enableEncryptionWithCustomerManagedKey: true,
+    allowCreateOperation: true,
+    additionalCreateRequestTemplates: {
+      'text/plain': 'Hello'
+    }
+  });
+
+  expect(stack).toHaveResourceLike('AWS::ApiGateway::Method', {
+    HttpMethod: 'POST',
+    Integration: {
+      RequestTemplates: {
+        'application/json': 'Action=SendMessage&MessageBody=$util.urlEncode(\"$input.body\")',
+        'text/plain': 'Hello'
+      }
+    }
+  });
+});
+
+test('Construct can override default create request template type', () => {
```

Comment:
Did we miss override default Read and Delete templates?

Reply:
good eye! I also added the missing default overrides checks to the aws-apigateway-dynamodb test as well.